src/codegen.cpp

```diff
@@ -674,7 +674,7 @@ static inline jl_cgval_t remark_julia_type(const jl_cgval_t &v, jl_value_t *typ)
 static inline jl_cgval_t mark_julia_const(jl_value_t *jv)
 {
     jl_value_t *typ;
-    if (jl_is_datatype(jv) || jl_is_uniontype(jv) || jl_is_typector(jv))
+    if (jl_is_type(jv))
         typ = (jl_value_t*)jl_wrap_Type(jv);
```

Comment:
this should probably be gc-rooted for at least the duration of codegen, right (and the one in `type_of_constant` too)?